lightning/src/ln/router.rs

```diff
@@ -1864,4 +1868,672 @@ mod tests {
+		};
+
+		let fake_msghash = hash_to_message!(&zero_hash);
+		match router.handle_node_announcement(
+			&NodeAnnouncement {
+				signature: secp_ctx.sign(&fake_msghash, secret_key),
+				contents: unsigned_announcement.clone()
+		}) {
+			Ok(_) => panic!(),
+			Err(e) => assert_eq!(e.err, "Invalid signature from remote node")
+		};
+
+		unsigned_announcement.timestamp += 1000;
+		unsigned_announcement.excess_data.push(1);
+		msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+		let announcement_with_data = NodeAnnouncement {
+			signature: secp_ctx.sign(&msghash, secret_key),
+			contents: unsigned_announcement.clone()
+		};
+		// Return false because contains excess data.
+	 	match router.handle_node_announcement(&announcement_with_data) {
+			Ok(res) => assert!(!res),
+			Err(e) => panic!("{}", e.err)
+		};
+		unsigned_announcement.excess_data = Vec::new();
+
+		unsigned_announcement.timestamp = 0;
+		msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+		let outdated_announcement = NodeAnnouncement {
+			signature: secp_ctx.sign(&msghash, secret_key),
+			contents: unsigned_announcement.clone()
+		};
+	 	match router.handle_node_announcement(&outdated_announcement) {
+			Ok(_) => panic!(),
+			Err(e) => assert_eq!(e.err, "Update older than last processed update")
+		};
+	}
+
+	#[test]
+	fn handling_channel_announcements() {
+		let (secp_ctx, _, router) = create_router();
+		let node_1_privkey = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_2_privkey = &SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_1_privkey);
+		let node_id_2 = PublicKey::from_secret_key(&secp_ctx, node_2_privkey);
+		let node_1_btckey = &SecretKey::from_slice(&[40; 32]).unwrap();
+		let node_2_btckey = &SecretKey::from_slice(&[39; 32]).unwrap();
+
+		let mut unsigned_announcement = UnsignedChannelAnnouncement {
+			features: ChannelFeatures::supported(),
+			chain_hash: genesis_block(Network::Testnet).header.bitcoin_hash(),
+			short_channel_id: 0,
+			node_id_1,
+			node_id_2,
+			bitcoin_key_1: PublicKey::from_secret_key(&secp_ctx, node_1_btckey),
+			bitcoin_key_2: PublicKey::from_secret_key(&secp_ctx, node_2_btckey),
+			excess_data: Vec::new(),
+		};
+
+		let channel_key = NetworkMap::get_key(unsigned_announcement.short_channel_id,
+						    unsigned_announcement.chain_hash);
+
+		let mut msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+		let valid_announcement = ChannelAnnouncement {
+			node_signature_1: secp_ctx.sign(&msghash, node_1_privkey),
+			node_signature_2: secp_ctx.sign(&msghash, node_2_privkey),
+			bitcoin_signature_1: secp_ctx.sign(&msghash, node_1_btckey),
+			bitcoin_signature_2: secp_ctx.sign(&msghash, node_2_btckey),
+			contents: unsigned_announcement.clone(),
+		};
+
+		// So the channel doesn't exist on chain here. Lookup returns ChainError::NotSupported
```

Comment:
This seems like a TODO, no? You'll probably want to build a test_util type like TestChannelMonitor where you can just set the return value in advance, see https://github.com/rust-bitcoin/rust-lightning/blob/master/lightning/src/util/test_utils.rs#L51